### PR TITLE
Add initial version of a DDD Starter Modelling Process

### PR DESCRIPTION
**Pull request by cakper**
Hello everyone! Together with @NTCoding we've put together an initial version of the starter process documentation. We would appreciate any feedback that you can give us - especially around the steps, resources and reasons why we think a particular step is required.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,275 @@
...
+![Context Map](resources/context-map.jpg)*Credit: Michael Plöd*
+
+#### Tools
+- [Dynamic Reteaming](https://leanpub.com/dynamicreteaming) 
+- [Pioneers, Settlers & Town Planners](http://wardleypedia.org/mediawiki/index.php/Pioneers_settlers_town_planners)
+- [Team Topologies](https://teamtopologies.com/)
+- [Visualising Sociotechnical Architecture with Context Maps](https://speakerdeck.com/mploed/visualizing-sociotechnical-architectures-with-context-maps)
+
+#### Who to Involve
+- Human Resources
```
Comment (yellowbrickc) [inline README.md:197]: HR is not involved in the whole process but here. Why? Legal and Compliance, etc are also needed, they are still not on this paper.
What about scrum master? Should they not be involved in everything regarding people and teams?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,275 @@
...
+- Software Architects
+- Product Managers
+- Senior Leadership
```
Comment (yellowbrickc) [inline README.md:200]: Who is now senior leadership, compared with technical leadership? What have the architects to do with all this?  No "architect" title guarantees good leadership...

**Comment by NTCoding** [comment]
> I hope, my feedback does not feel rude, it is not my intent at all! I think we have a totally different view regarding organisations and some parts of this paper have woken a few bad memories in me 😆

This is good. Including as many diverse perspectives as possible will help us to avoid bling spots and giving out the wrong messages.

Thank you, and please continue to share your valuable feedback whenever you feel inclined.

**Comment by selketjah** [comment]
This morning someone asked how they could start introducing DDD into their company if they were the only one interested in it and had a very beginners level (not feeling very confident) and Nick mentioned this starters kit.

I think this is a great starters kit for a team or company that is genuinely interested in introducing DDD into their company. I am not convinced that this is an helpful starters kit for people like the attendee we had this morning.

I think that would be different kind of starters kit, maybe not a starters kit but just tips and tricks such as:
- don't talk about bounded contexts or event storing or value objects
- ask in the next meeting with a user or business expert if they would be willing to indulge a little experiment to help you process their information and just write some post-its and put them on the wall while they talk

I am not sure, I feel it was more this type of advice they were looking for and I think that will also be valuable for beginners

File: README.md [hunk 3]
```diff
@@ -1,2 +1,275 @@
...
+Steps 7 (Define) and 8 (Code) can occur concurrently. This may happen when you are coding a bounded context, and the insights you get from writing code make you change the high-level design.
+
+## The Process
+This process is composed of 8 steps which are introduced below:
+
+### 1. Align
+Align our focus with the organisations' business model, the needs of its users, and its short, medium, and long-term goals.
+
+Every decision we take regarding the architecture, the code, or the organisation has business consequences. In order to design, build, and evolve software systems most effectively, our decisions need to create the optimal business impact, which can only be achieved if we are aligned to the business goals.
+
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply README.md:67]: Change incorporated during mob.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,275 @@
...
+- Senior Management
+- Technology Leadership
+
+### 2. Discover
+Discover the domain visually and collaboratively.
+
+Spreading domain knowledge through the whole team will create the shared understanding. It enables the developers to build a software system aligned to the team and more flexible to incorporate future business changes. 
+
+Ensuring that domain knowledge is spread across the whole team enables its members to contribute ideas for improving the product.
+
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply README.md]: Incorporated during mob session.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,275 @@
...
+### 3. Decompose
+Decompose the domain into sub-domains - loosely-coupled parts of the domain.
+
+We decompose a large problem domain into sub-domains for a few key reasons:
+- reduced cognitive load, so that we can reason about parts of the domain independently,
+- give development teams autonomy, so that they can work on separate parts of the solution,
+- identifying loose-coupling and high-cohesion in the domain which carries over to our software architecture and team structure.
+
+As a starting point, we recommend [carving up your event storm](resources/event-storm-bounded-context.jpg) into sub-domains.
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply README.md:130]: Change incorporated during mob session.

File: README.md [hunk 6]
```diff
@@ -1,2 +1,275 @@
...
+- [Business Capability Modelling](https://www.slideshare.net/trondhr/from-capabilities-to-services-modelling-for-businessit-alignment-v2)
+- [Design Heuristics](https://www.dddheuristics.com/)
+- [EventStorming with sub-domains](https://www.eventstorming.com/)
+- [Independent Service Heuristics](https://github.com/TeamTopologies/Independent-Service-Heuristics)
+
+#### Who to Involve
+- Domain Experts
+- Product Managers
+- Software Architects
+- Software Engineers
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply README.md]: Incorporated changes during mob.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,275 @@
...
+- Technology Leadership
+
+### 6. Organise
+Organise autonomous teams that are optimised for fast flow and aligned with context boundaries.
+
+We need to organise teams to have autonomy, clear goals and sense of purpose. In order to do that we need to take into account organisational constraints, so that we can organise teams for fast flow. 
+
+We can optimise how people collaborate with each other if we align teams with context boundaries. In order to right-size the teams we need to take into account available talent, cognitive load, communication overhead, and bus factor.
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply README.md]: Changes incorporated during mob session.

**Comment by NTCoding** [comment]
@selketjah 

As discussed during our mini-mob call. We will set up a meetup to identify techniques for adopting DDD in a gradual way.

**Comment by cakper** [comment]
I like how you reworked `Who to Involve` section - makes it clearer now and we can avoid calling out specific positions.

**Comment by trondhjort** [comment]
Hi,

Sorry for the late response. Great stuff this, even though I'm a bit worried that people with copy it whole-cloth and end up over-designing. Let's hope not.

I have a couple of comments:

- Step 1: You mention user need in the process illustration, but can't see it mentioned in the description. I think User Story Mapping would fit perfectly there, done in collaboration with biz and UX.

- Step 2: I think the user perspective is also often missed here, and find that user story mapping is also beneficial here to make all agree on what we would like to support when (slices in the story map).

- Step 4: Simon Brown has some extension to his C4 model which I have used and like, Dynamic Diagram (https://c4model.com/#SupplementaryDiagrams)

That's it. Thank you for including Business Capability Modelling.

**Pull request merged**